### DISCUSSION TITLE
What is the typing for next.js middleware config?

### DISCUSSION THREAD

**Discussion by RaenonX** [vercel/next.js#60092] [Help] [answered]
### Summary

I would like to type the `config` here:
```typescript
// middleware.ts
export const config = {
  matcher: ['/((?!api|account|_next|.*\\..*).*)'],
};
```

**Comment by rmnilin** [comment, marked as answer]
```ts
import type { MiddlewareConfig } from "next/server";
```

As I can see, it's not documented, and I was able to find only [one place](https://github.com/vercel/next.js/blob/9a1cd356dbafbfcf23d1b9ec05f772f766d05580/test/production/middleware-typescript/app/middleware.ts) where it was used directly to type middleware config in the entire Vercel codebase, and that with `satisfies` for some questionable reason. But it's there 😬

**Reply by RaenonX** [reply]
This is good enough, thanks! I agree that instead of `satisfies` it probably can just type it 😄
